### PR TITLE
feat: add helper to get the Python listing

### LINKED ISSUE #855: Reading Python classifiers from pyproject.toml
### How would this feature be useful?

We could convert python classifiers to a Python version list.

### Describe the solution you'd like

Here's the current way to do it:

```python
ALL_PYTHONS = [
    c.split()[-1]
    for c in nox.project.load_toml("pyproject.toml")["project"]["classifiers"]
    if c.startswith("Programming Language :: Python :: 3.")
]
```

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by henryiii**
Close #855.

### CODE REVIEW

File: nox/project.py [hunk 1]
```diff
@@ -67,3 +78,52 @@ def _load_script_block(filepath: Path) -> dict[str, Any]:
...
+        if from_classifiers:
+            return from_classifiers
+        raise ValueError('No Python version classifiers found in "project.classifiers"')
+
+    requires_python_str = pyproject.get("project", {}).get("requires-python", "")
+    if not requires_python_str:
+        raise ValueError('No "project.requires-python" value set')
+
+    for spec in packaging.specifiers.SpecifierSet(requires_python_str):
+        if spec.operator in {">", ">="}:
```
Comment (henryiii) [inline nox/project.py:121]:
```suggestion
        if spec.operator in {">", ">=", "~="}:
```

File: docs/config.rst [hunk 2]
```diff
@@ -466,6 +466,13 @@ class.
     :members:
     :undoc-members:
 
+The project helpers
```
Comment (theacodes) [inline docs/config.rst:469]: maybe "The project.toml helpers"? "project" is a bit generic on its own

File: nox/project.py [hunk 3]
```diff
@@ -67,3 +78,52 @@ def _load_script_block(filepath: Path) -> dict[str, Any]:
         for line in matches[0].group("content").splitlines(keepends=True)
     )
     return tomllib.loads(content)
+
+
+def python_list(
```
Comment (theacodes) [inline nox/project.py:83]: Maybe `python_versions`?

**Review by theacodes — COMMENTED** [review]
Mostly looks good, just some nitpicks about naming

**Comment by cjolowicz** [comment]
I was surprised that `max_version` determines from where we read the Python versions.

**Comment by cjolowicz** [comment]
Makes sense. Another approach would be to always look at Trove classifiers first, and fall back to `requires_python` if no classifiers are found. The `max_version` parameter would then just do what it says on the tin: apply an upper bound. (If it's not specified and we can't determine the upper bound from pyproject.toml, we could either bail out or apply our notion of what Python's current version is.) I don't feel strongly about this, though. Happy for this to go in 👍

**Comment by henryiii** [comment]
That's exactly what I originally tried, actually, but then I would have had to implement filtering, and why would you ever want to filter the classifiers using this parameter? What would be more likely to happen is someone would forget to update this bound when adding the classifier. Also, having it be "auto" means you'd have to pick what to do if classifiers and requires-python didn't match.

**Comment by ews-ffarella** [comment]
Hi!
Would that work with `Programming Language :: Python :: 3.13t`.
Or is this a bad classifiers?

**Comment by henryiii** [comment]
You can add to this list or process it, though. Including adding PyPy, etc.

**Pull request merged**